test/data_aggregator.test.ts

```diff
@@ -92,72 +88,90 @@ describe('DataAggregator', () => {
-            jest.spyOn(exchangeAdapter, 'fetchTicker').mockImplementation(async () => {
+        it('should throw if fetchAllPrices throws due to no successful tickers', async () => {
+          for (const priceSource of Array.from(dataAggregator.priceSources)) {
+            jest.spyOn(priceSource, 'fetchWeightedPrice').mockImplementation(async () => {
               throw Error('foo')
             })
           }
           await expect(async () => dataAggregator.currentPrice()).rejects.toThrow(
-            'All ticker requests failed'
+            'All price requests failed'
           )
         })
 
-        it('should throw if fetchAllTickers throws for any reason', async () => {
-          jest.spyOn(dataAggregator, 'fetchAllTickers').mockImplementation(async () => {
+        it('should throw if fetchAllPrices throws for any reason', async () => {
+          jest.spyOn(dataAggregator, 'fetchAllPrices').mockImplementation(async () => {
             throw Error('foo')
           })
           await expect(async () => dataAggregator.currentPrice()).rejects.toThrow('foo')
         })
       })
 
-      describe('cross-checks for tickers', () => {
+      describe('cross-checks for weighted prices', () => {
         beforeEach(() => {
           jest
-            .spyOn(dataAggregator, 'fetchAllTickers')
-            .mockImplementation(async () => Promise.resolve(testTickers))
+            .spyOn(dataAggregator, 'fetchAllPrices')
+            .mockImplementation(async () => Promise.resolve(testWeightedPrices))
         })
 
-        it('asks should not deviate more than askMaxPercentageDeviation', () => {
-          expect(() =>
-            aggregators.crossCheckTickerData(testTickerArray[8], dataAggregator.config)
-          ).toThrow('Max ask price cross-sectional deviation too large')
-        })
-
-        it('bids should not deviate more than bidMaxPercentageDeviation', () => {
-          expect(() =>
-            aggregators.crossCheckTickerData(testTickerArray[9], dataAggregator.config)
-          ).toThrow('Max bid price cross-sectional deviation too large')
+        it('prices should not deviate more than maxPercentageDeviation', () => {
+          const prices: WeightedPrice[] = [
+            { price: new BigNumber(0.7), weight: new BigNumber(1000) },
+            { price: new BigNumber(1.0), weight: new BigNumber(1000) },
+            { price: new BigNumber(1.3), weight: new BigNumber(1000) },
+          ]
+          expect(() => aggregators.crossCheckPriceData(prices, dataAggregator.config)).toThrow(
+            'Max price cross-sectional deviation too large'
+          )
         })
 
-        it('no exchange should have a volume share bigger than maxExchangeVolumeShare', () => {
-          expect(() =>
-            aggregators.crossCheckTickerData(testTickerArray[10], dataAggregator.config)
-          ).toThrow('The volume share of one exchange is too large')
+        // TODO(pedro-clabs): rename to maxSourceWeightShare?
```

Comment:
Is there any reason to not include this renaming in this PR? Seems like a more fitting name now that we're talking about Price Sources rather than Exchanges.

Reply:
The rationale was not breaking the config interface, so that we could have an oracle build that'd take both config types. It's no longer relevant as there are other interface breaks now. Renamed.